src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -782,6 +799,38 @@ namespace AirLoopHVACDOAS {
 
                 thisDOAS.m_AirLoopDOASNum = AirLoopDOASNum - 1;
                 state.dataAirLoopHVACDOAS->airloopDOAS.push_back(thisDOAS);
+
+                // ensure the inlet node should be an outdoor air node
+                bool OAnodeFound = false;
+                for (int OutsideAirNodeNum = 1; OutsideAirNodeNum <= state.dataOutAirNodeMgr->NumOutsideAirNodes; ++OutsideAirNodeNum) {
+                    if (thisDOAS.m_InletNodeNum == state.dataOutAirNodeMgr->OutsideAirNodeList(OutsideAirNodeNum)) {
+                        OAnodeFound = true;
+                        break;
+                    }
+                }
```

Comment:
There is a function for this:
`if(!OutAirNodeManager::CheckOutAirNodeNumber(state, thisDOAS.m_InletNodeNum) {`